libq/hash.h

```diff
@@ -44,5 +44,6 @@ int hash_multiple_file_at_cb(
 char *hash_file_at_cb(int pfd, const char *filename, int hash_algo, hash_cb_t cb);
 #define hash_file(f, h) hash_file_at_cb(AT_FDCWD, f, h, NULL)
 #define hash_file_at(fd, f, h) hash_file_at_cb(fd, f, h, NULL)
+char * hash_from_string(char *str,size_t len);
```

Comment:
This function now does hard-forced MD5 hashing, you'd expect that in the name.  But why not add an argument `int hash_algo` that will allow it to produce any hash?  You'd get something like: ```
char *hash_string(char *str, int len, int hash_algo);
```
then when len < 0, you assume str is NULL-terminated (to be able to compute the hash for the empty string).